src/test/java/seedu/address/model/project/DeadlineTest.java

```diff
@@ -0,0 +1,37 @@
+package seedu.address.model.project;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class DeadlineTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Deadline(null));
+    }
+
+    @Test
+    public void constructor_invalidDeadline_throwsIllegalArgumentException() {
+        String invalidDeadline = "";
+        assertThrows(IllegalArgumentException.class, () -> new Deadline(invalidDeadline));
+    }
+
+    @Test
+    void isValidDeadline() {
+        // null deadline
+        assertThrows(NullPointerException.class, () -> seedu.address.model.project.Deadline.isValidDeadline(null));
+
+        // invalid deadline
+        assertFalse(Deadline.isValidDeadline("")); // empty string
+        assertFalse(Deadline.isValidDeadline(" ")); // spaces only
+        assertFalse(Deadline.isValidDeadline("Testing")); // string of text
+        assertFalse(Deadline.isValidDeadline("27 October 2022")); // date in full
```

Comment:
May want to consider inputs such as dd-mm-yyyy as more suitable tests

Reply:
`toString()` method can be tested as well in order to see if the desired output is obtained